src/main/java/com/sendgrid/Client.java

```diff
@@ -277,23 +278,19 @@ public Response delete(Request request) throws URISyntaxException, IOException {
 	}
 
 	private Response executeApiCall(HttpRequestBase httpPost) throws IOException {
-		CloseableHttpResponse serverResponse = null;
-		Response response = new Response();
+		CloseableHttpResponse serverResponse = httpClient.execute(httpPost);
```

Comment:
You are correct that this doesn't fix the error, and could result in some exceptional states not being caught, but I can imagine some sequence of events where serverResponse.close() isn't called with the existing code. Give me a bit to think about that.

With this finalizer as long as the client nulls their reference to Class SendGrid or the class goes out of scope the port should be released. I have tested this on my own machine by nulling my reference, forcing garbage collection and then busy waiting to keep the process open.

```java
	@Override
	public void finalize() throws Throwable {
		try {
			this.httpClient.close();
		} catch(IOException e) {
			throw new Throwable(e.getMessage());
		} finally {
			super.finalize();
		}
	}
```